CMakeLists.txt

```diff
@@ -9,6 +9,17 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
 add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-unused-result -Werror)
 
+# Add Address Sanitiezed build type
+set(CMAKE_C_FLAGS_ASAN
+    "${CMAKE_C_FLAGS_DEBUG} -fsanitize=address -fno-omit-frame-pointer"
+    CACHE STRING "Flags used by the C compiler during address sanitizer builds."
+    FORCE )
+mark_as_advanced(
+    CMAKE_C_FLAGS_ASAN
+    CMAKE_EXE_LINKER_FLAGS_DEBUG
+    CMAKE_SHARED_LINKER_FLAGS_DEBUG
+    )
```

Comment:
Why `mark_as_advanced` means here?

Reply:
There is no need for these variables to be visible any cmake GUIs, so we can mark them as advanced. Actually `CMAKE_EXE_LINKER_FLAGS_DEBUG` and `CMAKE_SHARED_LINKER_FLAGS_DEBUG` will already be marked as advanced, so this could be compressed to `mark_as_advanced(CMAKE_C_FLAGS_ASAN)`